src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java

```diff
@@ -504,6 +549,24 @@ public final int batchMaxSize() {
         return getInt(BATCH_MAX_SIZE_CONFIG);
     }
 
+    // White space is significant for our batch delimiters but ConfigKey trims it out
+    // so we need to check the originals rather than using the normal machinery.
+    private String getOriginalString(final String key, final String defaultValue) {
+        return originalsStrings().getOrDefault(key, defaultValue);
+    }
+
+    public final String batchPrefix() {
+        return getOriginalString(BATCH_PREFIX_CONFIG, "");
+    }
+
+    public final String batchSuffix() {
+        return getOriginalString(BATCH_SUFFIX_CONFIG, "\n");
```

Comment:
I originally had this as the empty string like with prefix but changed to the newline since the current batching configuration does include a trailing newline after the last record. This may not be the most intuitive for new usages but provides backwards compatibility for existing connectors, so I'll defer to maintainers on which is preferred here.

Reply:
One shouldn't need `originalsStrings` here as `getString` can return `null`. Also calling `getString` instead `originalsString` marks configuration parameter as "used". Thus something like this:
```suggestion
        String result = getString(BATCH_SUFFIX_CONFIG);
        return result != null ? result : "\n";
```

Reply:
It's not so much about the `null` but rather about totally valid delimiters like `\n` or `\t` getting `trim()`ed to the empty string. For example, if the user specifies `"batch.separator": "\n"` (or `\t` anything else subject to `trim()`), then `batchSeparator()` will return the empty string, breaking batching, since the parsing logic always calls `trim()` and `getString()` works only with the trimmed values. Fortunately, the parsing logic does also stash the original string values, and the `originalsStrings()` hack was something I [found](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/83c8b47f2732f4206760ee132e7e474547c990ea/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java#L786-L793) in the S3 connector from Confluent while looking at how other connectors worked around this limitation.

Reply:
You're right here, I didn't see fully how this trimming was used.